src/modules/dAngle/process.cpp

```diff
@@ -21,25 +22,131 @@ Module::ExecutionResult DAngleModule::process(ModuleContext &moduleContext)
         return ExecutionResult::Failed;
     }
 
-    // Ensure any parameters in our nodes are set correctly
-    calculateAngle_->keywords().set("Symmetric", symmetric_);
-    dAngleNormalisationExpression_->setExpression(
-        fmt::format("{} * value/sin(toRad(y))/sin(toRad(yDelta))", symmetric_ ? 1.0 : 2.0));
-    collectDistance_->keywords().set("RangeX", distanceRange_);
-    collectAngle_->keywords().set("RangeX", angleRange_);
-    collectDAngle_->keywords().set("RangeX", distanceRange_);
-    collectDAngle_->keywords().set("RangeY", angleRange_);
-    if (excludeSameMolecule_)
-        selectC_->keywords().set("ExcludeSameMolecule", ConstNodeVector<SelectProcedureNode>{selectA_});
-    else
-        selectC_->keywords().set("ExcludeSameMolecule", ConstNodeVector<SelectProcedureNode>{});
-
-    // Execute the analysis
-    if (!analyser_.execute({moduleContext.dissolve(), targetConfiguration_, name()}))
+    auto &processingData = moduleContext.dissolve().processingModuleData();
+
+    // Select site A
+    SiteSelector a(targetConfiguration_, a_);
+
+    // Select site B
+    SiteSelector b(targetConfiguration_, b_);
+
+    // Select site C
+    SiteSelector c(targetConfiguration_, c_);
+
+    // RDF(B-C)
+    auto [rBC, rABStatus] = processingData.realiseIf<Histogram1D>("rBC", name(), GenericItem::InRestartFileFlag);
+    if (rABStatus == GenericItem::ItemStatus::Created)
+        rBC.initialise(distanceRange_.x, distanceRange_.y, distanceRange_.z);
+
+    // Angle(A-B-C)
+    auto [aABC, aABCStatus] = processingData.realiseIf<Histogram1D>("aABC", name(), GenericItem::InRestartFileFlag);
+    if (aABCStatus == GenericItem::ItemStatus::Created)
+        aABC.initialise(angleRange_.x, angleRange_.y, angleRange_.z);
+
+    // DAngle(A-BC)
+    auto [dAngle, dAngleStatus] = processingData.realiseIf<Histogram2D>("dAngle", name(), GenericItem::InRestartFileFlag);
+    if (dAngleStatus == GenericItem::ItemStatus::Created)
+        dAngle.initialise(distanceRange_.x, distanceRange_.y, distanceRange_.z, angleRange_.x, angleRange_.y, angleRange_.z);
+
+    dAngle.zeroBins();
+    rBC.zeroBins();
+    aABC.zeroBins();
+
+    // Site statistics
+    auto nASelections = 1;
+    auto nAAvailable = a.sites().size(), nACumulative = a.sites().size();
+    auto nBSelections = nAAvailable;
+    auto nBAvailable = 0, nBCumulative = 0;
+    auto nCSelections = 0, nCAvailable = 0, nCCumulative = 0;
+
+    for (const auto &[siteA, indexA] : a.sites())
     {
-        Messenger::error("CalculateDAngle experienced problems with its analysis.\n");
-        return ExecutionResult::Failed;
+        for (const auto &[siteB, indexB] : b.sites())
+        {
+
+            if (siteB->molecule() != siteA->molecule())
+                continue;
+
+            nBCumulative++;
+            nCSelections++;
+            nBAvailable++;
```

Comment:
It's a tiny, tiny thing, but prefer prefix increment over postfix (not for speed, but for intent in the sense that "we don't want use the value, just increase it").
```suggestion
            ++nBCumulative;
            ++nCSelections;
            ++nBAvailable;
```